rust/arrow/src/array/equal.rs

```diff
@@ -1046,6 +1062,30 @@ impl PartialEq<FixedSizeBinaryArray> for Value {
     }
 }
 
+impl JsonEqual for UnionArray {
+    fn equals_json(&self, _json: &[&Value]) -> bool {
+        unimplemented!()
+    }
+}
+
+impl PartialEq<Value> for UnionArray {
+    fn eq(&self, json: &Value) -> bool {
+        match json {
+            Value::Array(json_array) => self.equals_json_values(&json_array),
+            _ => false,
+        }
+    }
+}
+
+impl PartialEq<UnionArray> for Value {
+    fn eq(&self, arrow: &UnionArray) -> bool {
+        match self {
```

Comment:
nit: these two `eq` methods are very similar, but in one, you match on `self` and then compare to the argument and in the other one, this is transposed. It would be nice to make these consistent.

Reply:
Actually I was just following the pattern here, I thought that they must both be needed for IPC.  I removed them.  I'll add them back when I get to implementing IPC if needed.